app/controllers/results_controller.rb

```diff
@@ -55,15 +53,11 @@ def load_check
     @check = Check.new(session_data)
   end
 
-  def set_early_result_type
-    session_data["early_result"]["type"] = "gross_income"
-  end
-
   def track_completed_journey(calculation_result)
-    if signed_in? && current_provider.present?
-      JourneyLoggerService.call(assessment_id, calculation_result, @check, current_provider.first_office_code, cookies)
-    else
-      JourneyLoggerService.call(assessment_id, calculation_result, @check, nil, cookies)
-    end
+    office_code = signed_in? && current_provider.present? ? current_provider.first_office_code : nil
+
+    # We have the method `clear_early_result` on the ChecksController, so if a user goes back or ever see the check/change answers screen this will create a new entry into CompletedUserJourney
+    # The intention is that we are not counting that as a continuation an early ineligible result but as a new journey
+    JourneyLoggerService.call(assessment_id, calculation_result, @check, office_code, cookies) unless @check.early_ineligible_result?
```

Comment:
you could introduce the condition early on like 20 so that this method is never executed for an early result ie. 

track_completed_journey(@model) unless @check.early_ineligible_result?

I think this makes the condition a bit less hidden and ever so slightly saves on computing time lol.

You may need to move the @check instance variable further up the show methods if you go this way.